src/include/duckdb/common/arrow/appender/append_data.hpp

```diff
@@ -32,7 +32,7 @@ struct ArrowAppendData {
 	// the buffers of the arrow vector
 	ArrowBuffer validity;
 	ArrowBuffer main_buffer;
-	ArrowBuffer aux_buffer;
+	ArrowBuffer aux_buffer[2];
```

Comment:
I think at this point it might be easier to make this a `vector<ArrowBuffer>` and create getters for the Validity, MainBuffer, Auxiliary and BufferLengths, where we populate the vector with the first three always, and only add the BufferLengths when it's requested (checking `arrow_buffers.size() == 3` to determine if it needs to be created)

Reply:
That also results in having less of this `(uint8_t *)append_data.validity.data();` duplicated all over the place, which is a nice bonus